react-components/src/metadata/admin/EditMetadataTemplate.js

```diff
@@ -174,38 +151,42 @@ class EditMetadataTemplate extends Component {
 
 const validateAttributes = attributes => {
     const attributesArrayErrors = [];
-    const _error = [];
 
     attributes.forEach((attr, attrIndex) => {
         const attrErrors = {};
         const name = attr.name;
 
         if (!name) {
             attrErrors.name = getMessage("required");
+            attrErrors.error = getMessage("errAttrHasErrors");
             attributesArrayErrors[attrIndex] = attrErrors;
         } else {
             const namesMatch = attr => (attr.name === name);
             if (attributes.slice(0, attrIndex).some(namesMatch) || attributes.slice(attrIndex + 1).some(namesMatch)) {
                 attrErrors.name = getMessage("errAttrNameMustBeUnique");
+                attrErrors.error = attrErrors.error ? getMessage("errAttrHasErrors") : getMessage("errAttrNameMustBeUnique");
                 attributesArrayErrors[attrIndex] = attrErrors;
             }
         }
 
         // Validate Enum values
         if (attr.type === "Enum") {
             if (!attr.values || attr.values.length < 1) {
-                // Setting an attrErrors.values["_error"] message allows the error to be displayed in the table header.
-                attrErrors.values = [];
-                attrErrors.values["_error"] = getMessage("errEnumValueRequired");
+                // Setting an attrErrors.values message allows the error to be displayed in the table header,
+                // but that component will have to check first if attrErrors.values is an array or not.
+                attrErrors.values = getMessage("errEnumValueRequired");
```

Comment:
Here's an example of how I think Formik's `validate` function is more flexible than `redux-form`, which requires an `_error` to be set in each field's validation object if you want to access that custom error in the field's component.

On the other hand, Formik passes the entire `errors` object to each field, so they can access whatever is set in this function.

Reply:
👍